src/main/java/Kate.java

```diff
@@ -2,142 +2,266 @@
 import java.util.ArrayList;
```

Comment:
Great Job!
Overall good coding standard and quality.
However, it may not be a good idea to do the entire project in just one file